src/toil/fileStores/accessReportingWrapper.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2020 Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod, ABCMeta
+from contextlib import contextmanager
+import logging
+import os
+from typing import Any, Union
+
+from toil.fileStores import FileID
+from toil.fileStores.abstractFileStore import AbstractFileStore
+
+
+logger = logging.getLogger(__name__)
+
+class AccessReportingWrapper:
```

Comment:
Why not make this a subclass of AbstractFileStore ?

Reply:
If it is a subclass of AbstractFileStore, it gets all the methods of AbstractFileStore, which are stubs that raise `NotImplementedError`. I *think* those would have higher priority than `__getattr__` when someone goes to call a method, so we'd end up calling those stubs and not the methods of the wrapped FileStore.

I haven't tested it that way, though.

I could also extend AbstractFileStore and implement all those methods with overrides that do the call on the wrapped file store. That would be more code, but probably simpler code.